src/dfx/src/lib/builders/motoko.rs

```diff
@@ -206,16 +178,13 @@ struct MotokoParams<'a> {
 impl MotokoParams<'_> {
     fn to_args(&self, cmd: &mut std::process::Command) {
         cmd.arg(self.input);
-        cmd.arg("-o").arg(self.output).arg("--idl");
+        cmd.arg("-o").arg(self.output);
         match self.build_target {
             BuildTarget::Release => cmd.args(&["-c", "--release"]),
             BuildTarget::Debug => cmd.args(&["-c", "--debug"]),
-            BuildTarget::Idl => cmd.arg("--idl"),
-            BuildTarget::StableTypes => cmd.arg("--stable-types"),
-        };
-        if matches!(self.build_target, BuildTarget::Release | BuildTarget::Debug) {
-            cmd.arg("--public-metadata").arg("candid:service");
         };
+        cmd.arg("--idl").arg("--stable-types");
+        cmd.arg("--public-metadata").arg("candid:service");
```

Comment:
I wonder if we should at least have a setting in dfx.json for that... Making something implicitly public by default seems wrong to me.

Reply:
Yeah, the hacky query method is already public, so the default should be public. Let me add a TODO to control this from dfx.json.